homeassistant/components/openrgb/__init__.py

```diff
@@ -0,0 +1,213 @@
+
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_CLIENT_ID, CONF_HOST, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import (
+    DEFAULT_CLIENT_ID,
+    DEFAULT_PORT,
+    DOMAIN,
+    ENTRY_IS_SETUP,
+    ORGB_DATA,
+    ORGB_DISCOVERY_NEW,
+    ORGB_TRACKER,
+    SERVICE_FORCE_UPDATE,
+    SERVICE_PULL_DEVICES,
+    SIGNAL_DELETE_ENTITY,
+    SIGNAL_UPDATE_ENTITY,
+    TRACK_INTERVAL,
+)
+from .helpers import orgb_entity_id
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_HOST): cv.string,
+                    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                    vol.Optional(CONF_CLIENT_ID, default=DEFAULT_CLIENT_ID): cv.string,
+                }
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the OpenRGB integration."""
+    conf = config.get(DOMAIN)
+    if conf is not None:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up OpenRGB platform."""
+    try:
+        orgb = OpenRGBClient(
+            entry.data[CONF_HOST],
+            entry.data[CONF_PORT],
+            name=entry.data[CONF_CLIENT_ID],
+        )
+    except ConnectionError as err:
+        _LOGGER.error("Connection error during integration setup. Error: %s", err)
+        return False
+
+    def connection_recovered():
+        if not hass.data[DOMAIN]["online"]:
+            _LOGGER.info(
+                "Connection reestablished to OpenRGB SDK Server at %s:%i",
+                entry.data[CONF_HOST],
+                entry.data[CONF_PORT],
+            )
+
+        hass.data[DOMAIN]["online"] = True
+        async_dispatcher_send(hass, SIGNAL_UPDATE_ENTITY)
+
+    def connection_failed():
+        if hass.data[DOMAIN]["online"]:
+            hass.data[DOMAIN][ORGB_DATA].comms.stop_connection()
+            _LOGGER.info(
+                "Connection lost to OpenRGB SDK Server at %s:%i",
+                entry.data[CONF_HOST],
+                entry.data[CONF_PORT],
+            )
+
+        hass.data[DOMAIN]["online"] = False
+        async_dispatcher_send(hass, SIGNAL_UPDATE_ENTITY)
+
+    hass.data[DOMAIN] = {
+        "online": True,
+        ORGB_DATA: orgb,
+        ORGB_TRACKER: None,
+        ENTRY_IS_SETUP: set(),
+        "entities": {},
+        "pending": {},
+        "connection_failed": connection_failed,
+        "connection_recovered": connection_recovered,
```

Comment:
You use a mix of magic strings and constants here. Best practice followed by other integrations would be to use constants for all of these.